artipie-main/src/test/java/com/artipie/conda/CondaS3ITCase.java

```diff
@@ -0,0 +1,340 @@
+        );
+        this.containers.assertExec(
+            "Failed to move condarc to /root", new ContainerResultMatcher(),
+            "mv", "/w/.condarc", "/root/"
+        );
+        this.containers.assertExec(
+            "Failed to set anaconda upload url",
+            new ContainerResultMatcher(),
+            "anaconda", "config", "--set", "url",
+            String.format(
+                "http://artipie:%s/%s/",
+                CondaS3ITCase.PORT,
+                CondaS3ITCase.REPO
+            ), "-s"
+        );
+        this.containers.assertExec(
+            "Failed to set anaconda upload flag",
+            new ContainerResultMatcher(),
+            "conda config --set anaconda_upload yes".split(" ")
+        );
+        this.containers.assertExec(
+            "Minio unpack failed", new ContainerResultMatcher(),
+            "tar xvf /w/minio-bin-20231120.txz -C /root".split(" ")
+        );
+        this.containers.assertExec(
+            "Failed to start Minio", new ContainerResultMatcher(),
+            "bash", "-c", "nohup /root/bin/minio server /var/minio &"
+        );
+        this.containers.assertExec(
+            "Failed to wait for Minio", new ContainerResultMatcher(),
+            "sleep", "2"
+        );
+        this.containers.assertExec(
+            "Failed to configure Minio alias", new ContainerResultMatcher(),
+            "/root/bin/mc alias set srv1 http://localhost:9000 minioadmin minioadmin".split(" ")
+        );
+        this.containers.assertExec(
+            "Failed to configure Minio bucket", new ContainerResultMatcher(),
+            "/root/bin/mc mb srv1/buck1 --region s3test".split(" ")
+        );
+        this.containers.assertExec(
+            "Failed to configure Minio access", new ContainerResultMatcher(),
+            "/root/bin/mc anonymous set public srv1/buck1".split(" ")
+        );
+        this.containers.assertExec(
+            "Login was not successful",
+            new ContainerResultMatcher(),
+            "anaconda login --username alice --password 123".split(" ")
+        );
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+        "noarch_glom-22.1.0.tar.bz2,glom/22.1.0/noarch,noarch",
+        "snappy-1.1.3-0.tar.bz2,snappy/1.1.3/linux-64,linux-64"
+    })
+    void canSingleUploadToArtipie(final String pkgname, final String pkgpath, final String pkgarch)
+        throws IOException {
+        this.containers.assertExec(
+            "",
+            new ContainerResultMatcher(new IsEqual<>(CondaS3ITCase.CURL_NOT_FOUND)),
+            "curl -f -kv http://minic:9000/buck1/my-conda/%s/repodata.json".formatted(pkgarch).split(" ")
+        );
+        this.containers.assertExec(
+            "",
+            new ContainerResultMatcher(new IsEqual<>(CondaS3ITCase.CURL_NOT_FOUND)),
+            "curl -f -kv http://minic:9000/buck1/my-conda/%s/%s".formatted(pkgarch, pkgname).split(" ")
+        );
```

Comment:
could you please add some description here (first parameter in `assertExec`) to make it easier to understand what is checked in the assertion? Goes to all `assertExec` calls